lib/seccomp-tools/dumper-pid.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'seccomp-tools/ptrace'
+
+module SeccompTools
+  # Dump seccomp-bpf using ptrace of process.
+  module PidDumper
+    module_function
+
+    # Main bpf dump function.
+    #
+    # @param [Integer] pid
+    #   Target process identifier.
+    # @param [Integer] limit
+    #   Number of filters to dump. Negative number for unlimited.
+    # @yieldparam [String] bpf
+    #   Seccomp bpf in raw bytes.
+    # @yieldparam [Symbol] arch
+    #   Architecture of the target process (always nil right now).
+    # @return [Array<Object>, Array<String>]
+    #   Return the block returned. If block is not given, array of raw bytes will be returned.
+    # @example
+    #   dump(pid1, 1)
+    #   #=> []
+    #   dump(pid2, 1) { |c| c[0, 10] }
+    #   #=> [" \x00\x00\x00\x00\x00\x00\x00\x15\x00"]
+    def dump(pid, limit, &block)
```

Comment:
I would prefer to add a function (`dump_by_pid`) in `module Dumper` instead of create a new module.